From fb5362f205b37c5060fcd764a7ed393abe4f2f3d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 27 Jul 2018 17:39:37 +0800
Subject: [PATCH 1/2] fix opcode not supported on mips32-linux

While build tests(`make check') on mips32-linux, there are
serial failures such as:
[snip]
| mips-wrsmllib32-linux-gcc  -meb -mabi=32 -mhard-float -c
-o atomic_incs-atomic_incs.o `test -f 'atomic_incs.c' || echo
'../../../valgrind-3.13.0/memcheck/tests/'`atomic_incs.c
| /tmp/ccqrmINN.s: Assembler messages:
| /tmp/ccqrmINN.s:247: Error: opcode not supported on this
processor: mips1 (mips1) `ll $t3,0($t1)'
| /tmp/ccqrmINN.s:249: Error: opcode not supported on this
processor: mips1 (mips1) `sc $t3,0($t1)'
[snip]

Since the following commit applied, it defines CLFAGS for mips32,
but missed to pass them to tests which caused the above failure
...
3e344c57f Merge in a port for mips32-linux
...

Upstream-Status: Submitted [https://bugs.kde.org/show_bug.cgi?id=396905]
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 helgrind/tests/Makefile.am    | 5 +++++
 memcheck/tests/Makefile.am    | 5 +++++
 none/tests/mips32/Makefile.am | 4 ++++
 3 files changed, 14 insertions(+)

diff --git a/helgrind/tests/Makefile.am b/helgrind/tests/Makefile.am
index ad1af191a..6209d35a7 100644
--- a/helgrind/tests/Makefile.am
+++ b/helgrind/tests/Makefile.am
@@ -214,6 +214,11 @@ check_PROGRAMS += annotate_rwlock
 endif
 
 AM_CFLAGS   += $(AM_FLAG_M3264_PRI)
+
+if VGCONF_PLATFORMS_INCLUDE_MIPS32_LINUX
+AM_CFLAGS   += $(AM_CFLAGS_MIPS32_LINUX)
+endif
+
 AM_CXXFLAGS += $(AM_FLAG_M3264_PRI)
 
 LDADD = -lpthread
diff --git a/memcheck/tests/Makefile.am b/memcheck/tests/Makefile.am
index 84e49405f..aff861a32 100644
--- a/memcheck/tests/Makefile.am
+++ b/memcheck/tests/Makefile.am
@@ -443,6 +443,11 @@ check_PROGRAMS += reach_thread_register
 endif
 
 AM_CFLAGS   += $(AM_FLAG_M3264_PRI)
+
+if VGCONF_PLATFORMS_INCLUDE_MIPS32_LINUX
+AM_CFLAGS   += $(AM_CFLAGS_MIPS32_LINUX)
+endif
+
 AM_CXXFLAGS += $(AM_FLAG_M3264_PRI)
 
 if VGCONF_PLATFORMS_INCLUDE_ARM_LINUX
diff --git a/none/tests/mips32/Makefile.am b/none/tests/mips32/Makefile.am
index d11591d45..602cd26f6 100644
--- a/none/tests/mips32/Makefile.am
+++ b/none/tests/mips32/Makefile.am
@@ -99,6 +99,10 @@ check_PROGRAMS = \
 	round_fpu64 \
 	fpu_branches
 
+if VGCONF_PLATFORMS_INCLUDE_MIPS32_LINUX
+AM_CFLAGS   += $(AM_CFLAGS_MIPS32_LINUX)
+endif
+
 AM_CFLAGS    += @FLAG_M32@
 AM_CXXFLAGS  += @FLAG_M32@
 AM_CCASFLAGS += @FLAG_M32@
-- 
2.17.1

